### ISSUE TITLE
max-nested-callbacks: Ignore/ponderate some functions names

### ISSUE THREAD

**Issue by nfroidure** [eslint/eslint#3152]
This rule is pretty useful but i'd like to be able to configure some function names for which the depth of nesting would not be increased.

I thought to this option because mocha tests are based on callback nesting and the limit is reached too quickly so i'd like to be able to ignore 'describe/after/before/afterEach/beforeEach' or better being able to give them a ratio (1/2).

It could look like:
```js
"max-nested-callbacks": [2, 3, {
  'describe': 1/2,
  'before': 1/2,
  'after': 1/2,
  'then': 2,
}]
```

The above would give a smaller weight for mocha functions but a higher one for promises since promises can dramatically reduce nested callback when well used.

I can try to implement it and PR the project if you agree.

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by lo1tuma** [comment]
Out of curiosity, why you don't simply disable this rule for your tests? That's how I do it in my projects.

**Comment by neemzy** [comment]
:+1:

**Comment by nfroidure** [comment]
@lo1tuma our tests are in the same directories than our code so folder scope .eslinrc is not possible.

Also, the idea of specifying rules override in every file bother me.

And finally, i like the idea of being able to customize / prioritize nested callbacks.

**Comment by Nasga** [comment]
:+1:

**Comment by xvw** [comment]
+1

**Comment by lo1tuma** [comment]
@nfroidure Thats why we going to plan supporting glob-pattern based configurations for ESLint 2.0.

**Comment by nzakas** [comment]
I'd recommend creating your own custom rule if this is the functionality you want. I don't see it as being general enough to be included in the core.
